### DISCUSSION TITLE
Questions about the model

### DISCUSSION THREAD

**Discussion by Wenbo11** [ANL-CEEESA/UnitCommitment.jl#6] [Help & FAQ] [answered]
Hey Alison,

I have a few questions regarding the coded SCUC model:
- To my best understanding of the code, the SCUC model does not incorporate the network constraints and contingency constraints at the very beginning. It then incrementally enforces the transmission constraints and contingencies via ISF in the optimize! function \[https://github.com/ANL-CEEESA/UnitCommitment.jl/blob/dev/src/model.jl#L36\]. Do I understand it correctly? If it is correct, I understand you put the contingencies as the post-processing but why you do not incorporate the **transmission constraints** at every beginning?
- Does the coded SCUC model strictly follows the appendix of this paper \[https://pubsonline.informs.org/doi/10.1287/ijoc.2020.0976 \].

Thank you so much!

Best,
Wenbo

**Comment by iSoron** [comment, marked as answer]
Hi Wenbo, thank you for reaching out.

* Currently, we don't add any transmission or n-1 security constraints ahead of time; they are all added lazily. In *[Transmission Constraint Filtering in Large-Scale Security-Constrained Unit Commitment](https://ieeexplore.ieee.org/abstract/document/8613085)*, we compare the approach implemented here against the approach you describe (which was also proposed in *[Security Constrained Unit Commitment Using Line Outage Distribution Factors](https://ieeexplore.ieee.org/document/7886335)*, by Diego A. Tejada-Arango, Pedro Sánchez-Martın and Andres Ramos). That being said, if you find a better strategy to enforce these constraints, that would be a nice contribution, and we could incorporate it into the package.
* The formulation implemented here is similar to the one in the paper, but not strictly the same. For example, in the paper we did not consider time-dependent start-up costs, but UC.jl can handle them. Over time, I would also expect the formulations to diverge even more.
